### PR TITLE
add tikv 4.0 ga

### PR DESCRIPTION
**Pull request by WT-Liu**
Signed-off-by: Wenting Liu <liuwenting@pingcap.com>

### CODE REVIEW

File: content/blog/tikv-4.0-ga.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+title: TiKV 4.0 GA Release
+date: 2020-06-18
+author: TiKV Authors
+---
+
+We are excited to announce the general availability of TiKV 4.0! So glad and proud that we’ve seen more than 1000 adopters use TiKV in production scenarios across multiple industries worldwide. Thanks to all users who have given us feedback, bug reports, and pull requests contributions over the past year, we’ve applied the lessons learned from these deployments to bring new features and better security that can support users’ growing demands.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:7]:
```suggestion
We are excited to announce the general availability of TiKV 4.0! We're so glad and proud to have seen more than 1000 adopters use TiKV in production scenarios across multiple industries worldwide. Thanks to all users who have given us feedback, bug reports, and pull requests contributions over the past year. We’ve applied the lessons learned from these deployments to bring new features and better security to support users’ growing needs.
```

File: content/blog/tikv-4.0-ga.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# New Features
+
+For the 4.0 release, our team has kept polishing TiKV’s stability and creating new features. We’d like to highlight the following features which brought TiKV better performance and functionality!
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:11]:
```suggestion
For the 4.0 release, our team has kept polishing TiKV’s stability and creating new features. We’d like to highlight the following features which have brought TiKV better performance and functionality.
```

File: content/blog/tikv-4.0-ga.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Follower Read**
+
+The Follower Read feature uses a series of load balancing mechanisms to offload read requests on the Raft leader onto its followers in a Region. It ensures the linearizability of single-row data reads, helps reduce the load on the Region leader and substantially enhances the throughput of the entire system.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:15]:
```suggestion
With follower read, TiKV uses a series of load balancing mechanisms to offload read requests from the Raft leader to its followers. It ensures the linearizability of single-row data reads, helps reduce the load on the region leader, and substantially enhances the throughput of the entire system.
```

File: content/blog/tikv-4.0-ga.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Titan**
+
+Titan is a RocksDB plugin for key-value separation. Compatible with all RocksDB features used by the current TiKV, it reduces write amplification by separating values from the long-structured merge-tree (LSM tree) and storing them independently. Besides, it improves range query performance and reduces its impact on write performance. Titan has reached general availability in TiKV 4.0.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:19]:
```suggestion
Titan is a RocksDB plugin for key-value separation. Compatible with all RocksDB features used by TiKV, it reduces write amplification by separating large values from the log-structured merge tree (LSM tree) and storing them independently. It improves range query performance and reduces its impact on write performance. Titan has previously been available as an experimental feature, and has reached general availability in TiKV 4.0. 
```

@yiwu-arbug is it correct that Titan only separates large values?

File: content/blog/tikv-4.0-ga.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+The Follower Read feature uses a series of load balancing mechanisms to offload read requests on the Raft leader onto its followers in a Region. It ensures the linearizability of single-row data reads, helps reduce the load on the Region leader and substantially enhances the throughput of the entire system.
+
+*   **Titan**
+
+Titan is a RocksDB plugin for key-value separation. Compatible with all RocksDB features used by the current TiKV, it reduces write amplification by separating values from the long-structured merge-tree (LSM tree) and storing them independently. Besides, it improves range query performance and reduces its impact on write performance. Titan has reached general availability in TiKV 4.0. 
+
+*   **Unified Thread Pool**
+
+Unified Thread Pool is a unified adaptive thread pool in TiKV to process read requests. It unifies the Point Get Readpool and the Coprocessor Readpool to give priority to small requests, which improves resource utilization and predictably limit the impact of large queries on small requests.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:23]:
```suggestion
TiKV's unified thread tool is a unified adaptive thread pool in for processing read requests. It unifies the point-get read pool and the coprocessor read pool to give priority to small requests, which improves resource utilization and limits the impact of large queries on small requests.
```

File: content/blog/tikv-4.0-ga.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Load-based Splitting** 
+
+Load-based Splitting enables a Region to be automatically split into several small Regions when the Region remains a read hotspot continuously. After splitting, these small Regions can be evenly distributed to different TiKV nodes. Load-based Splitting solves the hot spots caused by the uneven distribution of Regions, such as full table scan and index query of small tables.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:27]:
```suggestion
Load-based splitting enables a region to be automatically split into several small regions when it is a read hotspot. After splitting, the new regions are evenly distributed to different TiKV nodes. Load-based splitting solves the hot spots caused by an uneven distribution of regions, such as full table scan and index query of small tables.
```

File: content/blog/tikv-4.0-ga.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+# Security
+
+To ensure TiKV is protected from internal and external threats, our team has been working hard on data protection. TiKV completed [a third-party security assessment](https://tikv.org/blog/tikv-pass-security-audit/) in March and this assessment of the TiKV scope, commissioned by CNCF and executed by Cure53 concludes with generally positive results. Besides, we have made TiKV 4.0 reach a higher level of data security for TiKV clusters deployed in cloud environments by:
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:31]:
```suggestion
To ensure TiKV is protected from internal and external threats, our team has been working hard on data protection. TiKV completed [a third-party security assessment](https://tikv.org/blog/tikv-pass-security-audit/) in March. This assessment of the TiKV scope, commissioned by CNCF and executed by Cure53, concluded with generally positive results. We have also improved data security for TiKV clusters deployed in cloud environments by implementing:
```

File: content/blog/tikv-4.0-ga.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Encryption at rest**
+
+TiKV 4.0 encrypts data stored at rest by default to ensure data reliability and security. Encryption at rest prevents unauthorized access to the unencrypted data by ensuring the data is encrypted when on disk to provide data protection for stored data.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:35]:
```suggestion
TiKV 4.0 encrypts data stored at rest by default to ensure data reliability and security. Encryption at rest prevents unauthorized access to data by ensuring the data is encrypted when on disk.
```

File: content/blog/tikv-4.0-ga.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **TLS in the HTTP(status) port**
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:37]:
```suggestion
*   **TLS in the HTTP (status) port**
```

File: content/blog/tikv-4.0-ga.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+TiKV 4.0 supports TLS in the HTTP (status) port and dynamically updates the certificate online, making it safe to fetch TiKV’s internal status through the HTTP port.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:39]:
```suggestion
TiKV 4.0 supports TLS in the HTTP (status) port and dynamically updates the certificate online, making it safer to fetch TiKV’s internal status through the HTTP port. 
```

File: content/blog/tikv-4.0-ga.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **TLS in the HTTP(status) port**
+
+TiKV 4.0 supports TLS in the HTTP (status) port and dynamically updates the certificate online, making it safe to fetch TiKV’s internal status through the HTTP port. 
+
+# What’s next
+
+Going GA means that TiKV 4.0 is ready for usage in production environments, but there is still much more to come for us. We’d love to listen to feature requests from the community. Currently, the best way to make feature requests is to [open an issue](https://github.com/tikv/tikv/issues/new?template=feature-request.md) on GitHub. The TiKV team will surely take that information to heart when driving this project forward.
+
+# Collaborate with TiKV Community
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:45]:
```suggestion
# Collaborate with the TiKV Community
```

File: content/blog/tikv-4.0-ga.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+TiKV 4.0 encrypts data stored at rest by default to ensure data reliability and security. Encryption at rest prevents unauthorized access to the unencrypted data by ensuring the data is encrypted when on disk to provide data protection for stored data.
+
+*   **TLS in the HTTP(status) port**
+
+TiKV 4.0 supports TLS in the HTTP (status) port and dynamically updates the certificate online, making it safe to fetch TiKV’s internal status through the HTTP port. 
+
+# What’s next
+
+Going GA means that TiKV 4.0 is ready for usage in production environments, but there is still much more to come for us. We’d love to listen to feature requests from the community. Currently, the best way to make feature requests is to [open an issue](https://github.com/tikv/tikv/issues/new?template=feature-request.md) on GitHub. The TiKV team will surely take that information to heart when driving this project forward.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:43]:
```suggestion
Going GA means that TiKV 4.0 is ready for use in production environments. There is still much more to come for us. We're planning future versions of TiKV, and we’d love to hear feature requests from the community. Currently, the best way to make feature requests is to [open an issue](https://github.com/tikv/tikv/issues/new?template=feature-request.md) on GitHub. The TiKV team will take that information to heart as we drive the project forward.
```

File: content/blog/tikv-4.0-ga.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Titan**
+
+Titan is a RocksDB plugin for key-value separation. Compatible with all RocksDB features used by the current TiKV, it reduces write amplification by separating values from the long-structured merge-tree (LSM tree) and storing them independently. Besides, it improves range query performance and reduces its impact on write performance. Titan has reached general availability in TiKV 4.0. 
+
+*   **Unified Thread Pool**
+
+Unified Thread Pool is a unified adaptive thread pool in TiKV to process read requests. It unifies the Point Get Readpool and the Coprocessor Readpool to give priority to small requests, which improves resource utilization and predictably limit the impact of large queries on small requests.
+
+*   **Load-based Splitting**
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:25]:
```suggestion
*   **Load-based splitting** 
```

File: content/blog/tikv-4.0-ga.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+*   **Follower Read**
+
+The Follower Read feature uses a series of load balancing mechanisms to offload read requests on the Raft leader onto its followers in a Region. It ensures the linearizability of single-row data reads, helps reduce the load on the Region leader and substantially enhances the throughput of the entire system.
+
+*   **Titan**
+
+Titan is a RocksDB plugin for key-value separation. Compatible with all RocksDB features used by the current TiKV, it reduces write amplification by separating values from the long-structured merge-tree (LSM tree) and storing them independently. Besides, it improves range query performance and reduces its impact on write performance. Titan has reached general availability in TiKV 4.0. 
+
+*   **Unified Thread Pool**
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:21]:
```suggestion
*   **Unified thread pool**
```

File: content/blog/tikv-4.0-ga.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+author: TiKV Authors
+---
+
+We are excited to announce the general availability of TiKV 4.0! So glad and proud that we’ve seen more than 1000 adopters use TiKV in production scenarios across multiple industries worldwide. Thanks to all users who have given us feedback, bug reports, and pull requests contributions over the past year, we’ve applied the lessons learned from these deployments to bring new features and better security that can support users’ growing demands.
+
+# New Features
+
+For the 4.0 release, our team has kept polishing TiKV’s stability and creating new features. We’d like to highlight the following features which brought TiKV better performance and functionality!
+
+*   **Follower Read**
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:13]:
```suggestion
*   **Follower read**
```

File: content/blog/tikv-4.0-ga.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
+---
+title: TiKV 4.0 GA Release
+date: 2020-06-18
+author: TiKV Authors
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:4]:
```suggestion
author: The TiKV authors
```

File: content/blog/tikv-4.0-ga.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+TiKV 4.0 supports TLS in the HTTP (status) port and dynamically updates the certificate online, making it safe to fetch TiKV’s internal status through the HTTP port. 
+
+# What’s next
+
+Going GA means that TiKV 4.0 is ready for usage in production environments, but there is still much more to come for us. We’d love to listen to feature requests from the community. Currently, the best way to make feature requests is to [open an issue](https://github.com/tikv/tikv/issues/new?template=feature-request.md) on GitHub. The TiKV team will surely take that information to heart when driving this project forward.
+
+# Collaborate with TiKV Community
+
+Most importantly, we’d like to thank our contributors who helped with this release. Whether you were a returning contributor or one of the many new folks we welcomed, thank you.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:47]:
```suggestion
Most importantly, we’d like to thank our contributors who helped with this release. Whether you were a returning contributor or one of the many new folks we welcomed, thank you!
```

File: content/blog/tikv-4.0-ga.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+
+Not a contributor yet? We’d love to help you get started if you’d like to get involved with the development and help drive forward the future of TiKV! You might be interested in tackling one of [these issues](https://github.com/tikv/tikv/issues?q=is%3Aopen+is%3Aissue+label%3Adifficulty%2Feasy). If you don’t know how to begin, just leave a comment and our team will help you out.
```
Comment (nrc) [inline content/blog/tikv-4.0-ga.md:49]:
```suggestion
Not a contributor yet? We’d love to help you get started! If you’d like to get involved with development and help drive the future of TiKV, you might be interested in tackling one of [these issues](https://github.com/tikv/tikv/issues?q=is%3Aopen+is%3Aissue+label%3Adifficulty%2Feasy). If you don’t know how to begin, just leave a comment and our team will help you out. 
```

**Comment by WT-Liu** [comment]
> Looks good, I only have minor grammar comments

Thanks, Nick! 😊

File: content/blog/tikv-4.0-ga.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+
+Load-based Splitting enables a Region to be automatically split into several small Regions when the Region remains a read hotspot continuously. After splitting, these small Regions can be evenly distributed to different TiKV nodes. Load-based Splitting solves the hot spots caused by the uneven distribution of Regions, such as full table scan and index query of small tables.
+
+# Security
+
+To ensure TiKV is protected from internal and external threats, our team has been working hard on data protection. TiKV completed [a third-party security assessment](https://tikv.org/blog/tikv-pass-security-audit/) in March and this assessment of the TiKV scope, commissioned by CNCF and executed by Cure53 concludes with generally positive results. Besides, we have made TiKV 4.0 reach a higher level of data security for TiKV clusters deployed in cloud environments by:
+
+*   **Encryption at rest**
+
+TiKV 4.0 encrypts data stored at rest by default to ensure data reliability and security. Encryption at rest prevents unauthorized access to the unencrypted data by ensuring the data is encrypted when on disk to provide data protection for stored data.
```
Comment (yiwu-arbug) [inline content/blog/tikv-4.0-ga.md:35]: This feature is not enabled by default.

Comment (WT-Liu) [reply content/blog/tikv-4.0-ga.md]: > This feature is not enabled by default.

Deleted it and thanks!

File: content/blog/tikv-4.0-ga.md [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+
+Load-based Splitting enables a Region to be automatically split into several small Regions when the Region remains a read hotspot continuously. After splitting, these small Regions can be evenly distributed to different TiKV nodes. Load-based Splitting solves the hot spots caused by the uneven distribution of Regions, such as full table scan and index query of small tables.
+
+# Security
+
+To ensure TiKV is protected from internal and external threats, our team has been working hard on data protection. TiKV completed [a third-party security assessment](https://tikv.org/blog/tikv-pass-security-audit/) in March and this assessment of the TiKV scope, commissioned by CNCF and executed by Cure53 concludes with generally positive results. Besides, we have made TiKV 4.0 reach a higher level of data security for TiKV clusters deployed in cloud environments by:
+
+*   **Encryption at rest**
+
+TiKV 4.0 encrypts data stored at rest to ensure data reliability and security. Encryption at rest prevents unauthorized access to the unencrypted data by ensuring the data is encrypted when on disk to provide data protection for stored data.
```
Comment (yiwu-arbug) [inline content/blog/tikv-4.0-ga.md:35]: should this be "TiKV 4.0 is able to encrypt data ..."?

File: content/blog/tikv-4.0-ga.md (hunk 19 above)
Comment (yiwu-arbug) [reply content/blog/tikv-4.0-ga.md]: should this be "TiKV 4.0 is able to encrypt data ..."? "Tikv 4.0 encrypts data" still sounds like it is on by default.

**Pull request merged**